test(linter): fix offset for partical source texts

Sysix opened the pull request:
Moved this PR into an own stack to fix the bug in https://github.com/oxc-project/oxc/pull/12545

camc314 commented:
### Merge activity

* **Jul 31, 10:35 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
test(linter): fix offset for partical source texts (#12594)

Moved this PR into an own stack to fix the bug in https://github.com/oxc-project/oxc/pull/12545
